runtime/lua/vim/lsp/util.lua

```diff
@@ -817,17 +817,21 @@ function M.convert_signature_help_to_markdown_lines(signature_help, ft, triggers
     if type(doc) == 'string' then
       signature.documentation = { kind = 'plaintext', value = doc }
     end
+    -- Add delimiter if there is documentation to display
+    if signature.documentation.value ~= '' then
+      contents[#contents + 1] = '---'
+    end
     M.convert_input_to_markdown_lines(signature.documentation, contents)
   end
   if signature.parameters and #signature.parameters > 0 then
     -- First check if the signature has an activeParameter. If it doesn't check if the response
-    -- had that property instead. Else just default to 0.
-    local active_parameter =
-      math.max(signature.activeParameter or signature_help.activeParameter or 0, 0)
```

Comment:
The spec still says "if `activeParameter` is omitted or the value lies outside the range of `signatures[activeSignature].parameters`, it defaults to 0 if the active signature has parameters". I think that this PR removes that?

Reply:
Or better put, if you're down for fixing https://github.com/neovim/neovim/issues/31368 that would be very welcome indeed :)